### DISCUSSION TITLE
Manage trashing on different drives ?

### DISCUSSION THREAD

**Discussion by ghost** [gokcehan/lf#982] [General]
Trashing / restoring with different partitions needs some modified mapping. Caz having the same trash-dir will move other partition stuff into homedir too. Does anyone know how to do this ? Need some code assistance.
Here's what I wanna do.
> Query mount location of current working dir
> `mkdir .trash` if it doesn't exist
> Move selected items into there

I'm using trash-cli btw.

**Comment by xezwary** [comment]
Not sure if the script I wrote is any good, and I haven't tested it properly, but is something like this what you're looking for?
```
cmd trashtest &{{
  set -f&&for i in $fx;do trashloc=$([[ "$(findmnt -n -o TARGET --target "${i%$(basename -- "$i")}")" == "/" ]]&&echo ~/.local/trash/||echo "$(findmnt -n -o TARGET --target "${i%$(basename -- "$i")}")/.trash/")&&[[ ! -d "$trashloc" ]]&&mkdir -p "$trashloc";f=$(basename "$i")&&[[ -e "$trashloc$f" || -L "$trashloc$f" ]]&&num=1&&until [[ ! -e "$trashloc$f" && ! -L "$trashloc$f" ]];do f="$(basename $i)~$num~"&&num=$((++num));done;mv -- "$i" "$trashloc$f";done;lf -remote "send $id reload"
}}
```
Personally I just use glib2.

**Reply by ghost** [reply]
Uh. Can you break it down / line escape it (\\) ? Can't understand. Tq for the snippet btw.

**Reply by xezwary** [reply]
Line escaped edition, with comments:
```
cmd trashtest &{{
  set -f
  for i in $fx;

    #Get trash location using findmnt
    do trashloc=$(
      [[ "$(findmnt -n -o TARGET --target "${i%$(basename -- "$i")}")" == "/" ]]&&echo ~/.local/trash/||echo "$(findmnt -n -o TARGET --target "${i%$(basename -- "$i")}")/.trash/"
    )
    [[ ! -d "$trashloc" ]]&&mkdir -p "$trashloc"

    #Get basename of file and append a number if a file with the same filename exists
    f=$(basename "$i")
    [[ -e "$trashloc$f" || -L "$trashloc$f" ]]&&num=1&&until [[ ! -e "$trashloc$f" && ! -L "$trashloc$f" ]];
      do f="$(basename $i)~$num~"&&num=$((++num))
    done
    mv -- "$i" "$trashloc$f";
  done;lf -remote "send $id reload"
}}
```

**Reply by ghost** [reply]
Tqvm. But trash-cli can do all of this. I just need to use `trash`, `trash-empty` &  `trash-restore`. But instead on home, wanna do it for other drives too (in their mounts).

**Reply by ghost** [reply]
@xezwary Also, can you mind lemme know how I can do trash confirmation from status bar ? (just like deletion on lf).

**Reply by xezwary** [reply]
You can use the % and use the read command to get a prompt:
```
cmd confirmtest %{{
  [[ $(echo "$fx"|wc -l) > 1 ]]&&echo "Delete $(echo "$fx"|wc -l) files? y/n "||echo "Delete $fx? y/n "&&read ans&&[[ ans == "y" ]]&&lf -remote "send $id reload"
}}
```
